### PR TITLE
Fix profile GC

### PR DESCRIPTION
**Pull request by santiagoaguiar**
When PROFILE_GC is set but HUNT_MEMORY_LEAKS is not, an uncaught is thrown because the `createThrottledSnapshot` function is not defined. 

This caused an uncaught on the auth0-mfa-api service: https://auth0.slack.com/archives/C04SSNEKT/p1538487947000100

This PR fix the issue by always creating the function, and adds a new env variable to opt in to get snapshots.

In the case of MFA we are interested in getting the GC stats, but not in getting heapdumps.

The change will also make that any service using `PROFILE_GC` will stop getting heapdumps until this new env variable is defined. As far as I know, no service is now using dumps created by GC blocks. In fact, we use `PROFILE_GC = true` without `HUNT_MEMORY_LEAKS` in most services (https://github.com/auth0/auth0-configuration/blob/ac7fae0a66450079cb59d8a19122630eaba6bdf3/hiera/common/us_amazon_west.json#L261).

Testing:
* Unit tests

### CODE REVIEW

File: lib/profiler.js [hunk 1]
```diff
@@ -128,11 +129,13 @@ Profiler.prototype.setupGCReporter = function setupGCReporter() {
...
-        this.createThrottledSnapshot('LONG_GC_PAUSE');
-      }, ms('5s'));
+      if (this.heapDumpOnGc) {
+        setTimeout(() => {
+          this.createThrottledSnapshot('LONG_GC_PAUSE');
+        }, ms('5s'));
+      }
 
-      this.agent.logger.info('long GC pause', {
+      this.agent.logger.error('long GC pause', {
```
Comment (santiagoaguiar) [inline lib/profiler.js]: Switching to error, we will want to get this messages if they occur.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "auth0-instrumentation",
-  "version": "2.18.0",
+  "version": "2.19.0",
```
Comment (santiagoaguiar) [inline package.json]: Already bumping, can send a separate PR if needed.

File: test/profiler.test.js [hunk 3]
```diff
@@ -55,4 +56,59 @@ describe('Profiler', function() {
...
+        clock = sinon.useFakeTimers(new Date(), 'setTimeout');
+      });
+
+      afterEach(() => {
+        clock.restore();
+      });
+
+      it ('does not takes snapshot', () => {
+        const profiler = getProfiler({ PROFILE_GC: true });
+        profiler.setupGCReporter();
```
Comment (pmalouin) [inline test/profiler.test.js]: Maybe we should not call `setupGCReporter()` a second time and rely on the invocation done in the constructor.

Applies to this test and the two others below.

File: lib/profiler.js [hunk 4]
```diff
@@ -10,13 +10,14 @@ const throttle = require('lodash.throttle');
 process.send = process.send || function () {};
 
 function Profiler(agent, pkg, env) {
-  this.huntMemoryLeaks = env.HUNT_MEMORY_LEAKS;
+  this.heapDumpOnGc = env.HUNT_LONG_GC;
```
Comment (pmalouin) [inline lib/profiler.js]: Thoughts:

`env.HUNT_LONG_GC` will only be taken into account if `env.PROFILE_GC` is true (`this.setupGCReporter()` won't be called).

I think it makes sense.

File: test/profiler.test.js (hunk 3 above)
Comment (santiagoaguiar) [reply test/profiler.test.js]: Thanks, fixed,

File: lib/profiler.js (hunk 1 above)
Comment (ziluvatar) [reply lib/profiler.js]: Really? Will you do any action if you get a "long GC pause" error? I'd rather like to leave it as info and add metric so if the service is interested they could configure an alert.

Comment (ziluvatar) [reply lib/profiler.js]: Actually there is a histogram metric at the beginning, so I'd leave it as info

File: test/profiler.test.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 const assert = require('assert');
 const sinon = require('sinon');
-
+var $require = require('proxyquire').noPreserveCache();
```
Comment (dafortune) [inline test/profiler.test.js]: const 🙏

File: package.json (hunk 2 above)
Comment (dafortune) [reply package.json]: Semantically it is a breaking change i think, before if you had:
PROFILE_GC + HUNT_MEMORY_LEAKS you got a head dump on LONG_GC_PAUSE (this could have been consider a bug too), now you don't get it because you also need HUNT_LONG_GC. 

I think this is closed to a bug fix than to a breaking change, minor seems and intermediate point, but fail to see it as a minor.

File: test/profiler.test.js (hunk 5 above)
Comment (santiagoaguiar) [reply test/profiler.test.js]: Done :)
